src/main/java/org/candlepin/service/exception/ServiceException.java

```diff
@@ -72,12 +71,12 @@ public ServiceException(String message, Throwable cause) {
         super(message, cause);
     }
 
-    public ServiceException setStatus(int status) {
-        this.status = status;
-        return this;
-    }
-
-    public int getStatus() {
-        return this.status;
+    /**
+     * Information needed for the generation of a translated message
+     *
+     * @return Object[] array of substitution data
+     */
+    public Object[] getDetails() {
```

Comment:
Hrmm... if the mapper is reliant on stuff like this, we're better off with the gigabranch that the original had. Definitely would like to see something slightly more involved like a function or consumer so we don't need to add generic abstractions like this to the exception API. Something closer to this:

```
mappings.add(ProductServiceException.class, pse -> {
  return Pair.of(
    Status.BAD_REQUEST,
    i18n.tr("Unable to retrieve product \"{0}\" from ProductService.", pse.getProductId()));
})
```

We can then also use `.getOrDefault` to wrap up the generic message stuff as well:

```
Pair<Status, String> result = this.mappings.getOrDefault(exception.getClass(), 
  e -> Pair.of(Status.BAD_REQUEST, i18n.get().tr("Unknown service exception"));

return this.getDefaultBuilder(..., result.getLeft());
```